### ISSUE TITLE
another error in installing nominatim

### ISSUE THREAD

**Issue by cellum** [twain47/Nominatim#249]
hi,
when I tried installing luxembourg, estonia or latvia then installing was successful and website works fine.. But if I try to install whole planet, I get error in indexing phase:
(nominatim and osm2pgsql are up-to-date, same version as in git, postgres - 9.3 and postgis - 2.1)
---
Done 1 in 43 @ 0.023256 per second - Rank 16 ETA (seconds): 5962810.000000
Done 1 in 44 @ 0.022727 per second - Rank 16 ETA (seconds): 6101480.000000
Done 1 in 44 @ 0.022727 per second - Rank 16 ETA (seconds): 6101480.000000
index_placex: UPDATE failed: ERROR:  GEOSBuffer: std::bad_alloc
CONTEXT:  PL/pgSQL function add_location(bigint,character varying,integer,integer[],integer,integer,geometry) line 51 at assignment
PL/pgSQL function placex_update() line 672 at assignment
ERROR: Error executing external command: /home/ubuntu/Nominatim/nominatim/nominatim -i -d nominatim -P 5433 -t 1 -r 5 -R 25
Error executing external command: /home/ubuntu/Nominatim/nominatim/nominatim -i -d nominatim -P 5433 -t 1 -r 5 -R 25
---

any suggestions?
thanks,
kalle

**Comment by lonvia** [comment]
Sounds like you are somehow out of memory. How much RAM and swap do you have and how is your postgres configured in terms of the parameters mentioned [here](https://wiki.openstreetmap.org/wiki/Nominatim/Installation#PostgreSQL)?  Could you also provide the exact version information as reported by `select postgis_full_version()`.

Also, please try to find out which OSM object by running first `./utils/setup.php --create-functions --enable-debug-statements` and then running `./utils/update.php --index`. That should immediately fail with the same error but you should get a lot of debug output in the postgres log file including a mention of either the place_id or osm_type/osm_id. If it is place_id, please look up the osm_id in the placex table.

**Comment by mtmail** [comment]
Also, the ETA of 6,101,480 seconds is 71 days so I the machine might be swapping to disc a lot or is underpowered.

**Comment by lonvia** [comment]
The ETA can't really be trusted because it is the first object being processed in rank 16.

**Comment by cellum** [comment]
more detailed error:
Starting rank 16
WARNING:  placex_update R 4064756
WARNING:  get_country_code, start: POINT(9.2860325 50.9121216)
CONTEXT:  PL/pgSQL function placex_update() line 106 at assignment
index_placex: UPDATE failed: ERROR:  record "relation" is not assigned yet
DETAIL:  The tuple structure of a not-yet-assigned record is indeterminate.
CONTEXT:  SQL statement "SELECT relation.members[i]"
PL/pgSQL function placex_update() line 120 at RAISE
ERROR: Error executing external command: /home/ubuntu/Nominatim/nominatim/nominatim -i -d nominatim -P 5433 -t 1 -r 5 -R 25


Postgis version is: 
--
POSTGIS="2.1.2 r12389" GEOS="3.4.2-CAPI-1.8.2 r3921" PROJ="Rel. 4.8.0, 6 March 2012" GDAL="GDAL 1.10.1, released 2013/08/26" LIBXML="2.9.1" LIBJSON="UNKNOWN" RASTER
--
Server is upcloud 2cpus, 8Gb RAM
Postgres parameters:
shared_buffers 2GB
maintenance_work_mem 400MB
work_mem 4MB
effective_cache_size 6GB
synchronous_commit = off
checkpoint_segments = 30
checkpoint_timeout = 15min
checkpoint_completion_target = 0.9
fsync = off
full_page_writes = off

and I assigned --osm2pgsql-cache 18000
I this too weak for installing nominatim for planet?

thanks,
kalle

**Comment by lonvia** [comment]
That relation looks familiar, I'm afraid. The same error was already reported in #214. Please try increasing `work_mem` to 100MB (you might have to restart postgresql after the change). You can continue the import process by running `./utils/setup.php --create-functions --index --create-search-indices`.

If it doesn't work you may try to disable waterway relations by adding the following at the beginning of the placex_update() function in sql/functions.sql (line 1278):

~~~
IF NEW.osm_type = 'R' and NEW.class = 'waterway' THEN
  RETURN NEW;
END IF;
~~~

Again, run `./utils/setup.php --create-functions --index --create-search-indices` to continue the import.

Be warned that your machine is rather weak for a planet-wide indexing, so things will be very, very slow but it should work. (These days I wouldn't recommend less than 32GB for a planet-wide import.)

**Comment by cellum** [comment]
thaks for advice.. I try these steps.

**Comment by lonvia** [comment]
Closed as duplicate of #214.
